text/0008-parachain-bootnodes-dht.md

```diff
@@ -0,0 +1,148 @@
+| **Start Date**  | 2023-07-14                                                                                  |
+| **Description** | Parachain bootnodes shall register themselves in the DHT of the relay chain                 |
+| **Authors**     | Pierre Krieger                                                                              |
+
+## Summary
+
+The full nodes of the Polkadot peer-to-peer network maintain a distributed hash table (DHT), which is currently used for full nodes discovery and validators discovery purposes.
+
+This RFC proposes to extend this DHT to be used to discover full nodes of the parachains of Polkadot.
+
+## Motivation
+
+The maintenance of bootnodes has long been an annoyance for everyone.
+
+When a bootnode is newly-deployed or removed, every chain specification must be updated in order to take the update into account. This has lead to various non-optimal solutions, such as pulling chain specifications from GitHub repositories.
+When it comes to RPC nodes, UX developers often have trouble finding up-to-date addresses of parachain RPC nodes. With the ongoing migration from RPC nodes to light clients, similar problems would happen with chain specifications as well.
+
+Furthermore, there exists multiple different possible variants of a certain chain specification: with the non-raw storage, with the raw storage, with just the genesis trie root hash, with or without checkpoint, etc. All of this creates confusion. Removing the need for parachain developers to be aware of and manage these different versions would be beneficial.
+
+Since the PeerId and addresses of bootnodes needs to be stable, extra maintenance work is required from the chain maintainers. For example, they need to be extra careful when migrating nodes within their infrastructure. In some situations, bootnodes are put behind domain names, which also requires maintenance work.
+
+Because the list of bootnodes in chain specifications is so annoying to modify, the consequence is that the number of bootnodes is rather low (typically between 2 and 15). In order to better resist downtimes and DoS attacks, a better solution would be to use every node of a certain chain as potential bootnode, rather than special-casing some specific nodes.
+
+While this RFC doesn't solve these problems for relay chains, it aims at solving it for parachains by storing the list of all the full nodes of a parachain on the relay chain DHT.
+
+Assuming that this RFC is implemented, and that light clients are used, deploying a parachain wouldn't require more work than registering it onto the relay chain and starting the collators. There wouldn't be any need for special infrastructure nodes anymore.
+
+## Stakeholders
+
+This RFC has been opened on my own initiative because I think that this is a good technical solution to a usuability problem that many people are encountering and that they don't realize can be solved.
```

Comment:
```suggestion
This RFC has been opened on my own initiative because I think that this is a good technical solution to a usability problem that many people are encountering and that they don't realize can be solved.
```